src/walletConnectProvider.ts

```diff
@@ -162,27 +161,19 @@ export class WalletConnectProvider implements IDappProvider {
             method: "erd_batch_sign",
             params
         });
-        if (!signatures) {
+        if (!signatures || !Array.isArray(signatures)) {
             Logger.error("signTransactions: Wallet Connect could not sign the transactions");
             throw new Error("Wallet Connect could not sign the transactions");
         }
 
-        if (Array.isArray(signatures)) {
-            if (transactions.length !== signatures.length) {
-                Logger.error("signTransactions: Wallet Connect could not sign the transactions. Invalid signatures.");
-                throw new Error("Wallet Connect could not sign the transactions. Invalid signatures.");
-            }
-
-            transactions.map((transaction, key: number) => 
-                transaction.applySignature(Signature.fromHex(signatures[key].signature), UserAddress.fromBech32(address))
-            );
-
-            return transactions;
+        if (transactions.length !== signatures.length) {
+            Logger.error("signTransactions: Wallet Connect could not sign the transactions. Invalid signatures.");
+            throw new Error("Wallet Connect could not sign the transactions. Invalid signatures.");
         }
 
-        transactions[0].applySignature(Signature.fromHex(signatures.signature), UserAddress.fromBech32(address));
-        
-        return transactions;
+        transactions.map((transaction, key: number) => 
```

Comment:
usually you would use .map to create another array, but here it simply goes over the tx. Wouldn't it be better to use forEach or for of?

Reply:
Fixed.